includes/core/DataTransferObjects/API/class-response.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Response DTO.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\DataTransferObjects;
+
+use WCPay\Core\Contracts\API\Response\Base_Response;
+use WP_Http_Cookie;
+
+/**
+ * DTO for parsing response from WCPay server API.
+ */
+final class Response extends Data_Transfer_Object implements Base_Response {
```

Comment:
```php
extends Data_Transfer_Object
```

I feel that this class is doing a bit too much to be categorized as DTO. To me, it is a strict object representing a response from the wcpay server. I think this is a core class that will be involved in many important business logic representations. Calling it `DTO` is devaluating it and we might have problems in future where the "DTO" label will influence our decision of not adding specific methods to it when needed and looking for other solutions. 

The focus of this PR is making the interaction with the server more structured (`Request` and `Response` classes). M worry is that by introducing the term "DTO" we'll bypass the necessary discussions of how we want the DTO to be implemented in our codebase. IMHO, what's currently in `Data_Transfer_Object` is not fully versatile for mass adoption so I would be worried that we'll fill the terminology slot and make things harder for us later.

With that in mind, I would suggest moving the methods from `Data_Transfer_Object` to the `Response`.

Reply:
Yes, the focus is on interaction between client and server, but I wanted to make a foundation for other parts of the plugin where you could potentially use DTO and create a class that could share common logic for all DTOs somewhere in the future.

But as you mentioned, the purpose of this PR is server interaction and I will rename all the DTOs, ValueObjects, and other things to represent client server interaction. DTOs will be replaced with Requests/Responses

Reply:
Since we are dealing with Requests and Responses, this class have been refactored and DTO abstract class is removed.